apps/core/lib/core/snitch/data/model/payment/card_payment.ex

```diff
@@ -0,0 +1,113 @@
+defmodule Core.Snitch.Data.Model.CardPayment do
+  @moduledoc """
+  CardPayment API and utilities.
+
+  `CardPayment` is a concrete payment subtype in Snitch. By `create/2`ing a
+  CardPayment, the supertype Payment is automatically created in the same
+  transaction.
+
+  > For other supported payment sources, see
+    `Core.Snitch.Data.Schema.PaymentMethod`
+  """
+  use Core.Snitch.Data.Model
+
+  @doc """
+  Creates both `Payment` and `CardPayment` records in a transaction for Order
+  represented by `order_id`.
+
+  * `payment_params` are validated using
+    `Core.Snitch.Data.Schema.Payment.changeset/3` with the `:create` action and
+    because `slug` and `order_id` are passed explicitly to this function,
+    they'll be ignored if present in `payment_params`.
+  * `card_params` are validated using
+  `Core.Snitch.Data.Schema.CardPayment.changeset/3` with the `:create` action.
+  """
+  @spec create(String.t(), non_neg_integer(), map, map) ::
+          {:ok, %{card_payment: Schema.CardPayment.t(), payment: Schema.Payment.t()}}
+          | {:error, Ecto.Changeset.t()}
+  def create(slug, order_id, payment_params, card_params) do
+    payment = struct(Schema.Payment, payment_params)
+    card_method = Model.PaymentMethod.get_card()
+
+    more_payment_params = %{
+      order_id: order_id,
+      payment_type: "ccd",
+      payment_method_id: card_method.id,
+      slug: slug
+    }
+
+    payment_changeset = Schema.Payment.changeset(payment, more_payment_params, :create)
+
+    Ecto.Multi.new()
+    |> Ecto.Multi.insert(:payment, payment_changeset)
+    |> Ecto.Multi.run(:card_payment, fn %{payment: payment} ->
+      all_card_params = Map.put(card_params, :payment_id, payment.id)
+      QH.create(Schema.CardPayment, all_card_params, Repo)
+    end)
+    |> Core.Repo.transaction()
+  end
+
+  @deprecated "This is dangerous as it allows changing the amount"
```

Comment:
Why should we allow updating amounts? Should we keep this or remove it?

Reply:
As we discussed last time; since this is not going to be used in the flow that we discussed so it's better if we remove it.

Reply:
If we are providing the option of partial payments, then the user might need to change the amount he/she wants to pay using the card option in that case this might be needed.

Reply:
Partial payments do not imply that the user should be able to edit the amount. I'm listing all scenarios where an order has more than 1 payment:
1. Order amount is `x` and a user has `k/100 * x` in his store credit. The user _can_ make up to two payments,
    - `k/100 * x` by store-credit
    - `(100 - k)/100 * x` by card/cash/whatever
2. The user has paid the complete order amount (possibly via multiple payments) AND the admin increases the order total. The user will have to make _at least one_ additional payment now for the outstanding amount.`